Tests/Model/ModelManagerTest.php

```diff
@@ -170,4 +175,118 @@ private function getMetadata($class, $isVersioned)
+        $associatedEntityClass = 'Sonata\DoctrineORMAdminBundle\Tests\Fixtures\Entity\AssociatedEntity';
+        $embeddedEntityClass = 'Sonata\DoctrineORMAdminBundle\Tests\Fixtures\Entity\Embeddable\EmbeddedEntity';
+        $modelManagerClass = 'Sonata\DoctrineORMAdminBundle\Model\ModelManager';
+
+        $object = new ContainerEntity(new AssociatedEntity(), new EmbeddedEntity());
+
+        $em = $this->getMockBuilder('Doctrine\ORM\EntityManager')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        /** @var \PHPUnit_Framework_MockObject_MockObject|ModelManager $modelManager */
+        $modelManager = $this->getMockBuilder($modelManagerClass)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getMetadata', 'getEntityManager'))
+            ->getMock();
+
+        $modelManager->expects($this->any())
+            ->method('getEntityManager')
+            ->will($this->returnValue($em));
+
+        $containerEntityMetadata = $this->getContainerEntityMetadata($containerEntityClass);
+        $associatedEntityMetadata = $this->getContainerEntityMetadata($associatedEntityClass);
+        $embeddedEntityMetadata = $this->getContainerEntityMetadata($embeddedEntityClass);
+
+        $modelManager->expects($this->any())->method('getMetadata')
+            ->will(
+                $this->returnValueMap(
+                    array(
+                        array($containerEntityClass, $containerEntityMetadata),
+                        array($embeddedEntityClass, $embeddedEntityMetadata),
+                        array($associatedEntityClass, $associatedEntityMetadata),
+                    )
+                )
+            );
+
+        /** @var ClassMetadata $metadata */
+        list($metadata, $lastPropertyName) = $modelManager
+            ->getParentMetadataForProperty($containerEntityClass, 'plainField');
+        $this->assertEquals($metadata->fieldMappings[$lastPropertyName]['type'], 'integer');
+
+        list($metadata, $lastPropertyName) = $modelManager
+            ->getParentMetadataForProperty($containerEntityClass, 'associatedEntity.plainField');
+        $this->assertEquals($metadata->fieldMappings[$lastPropertyName]['type'], 'string');
+
+        list($metadata, $lastPropertyName) = $modelManager
+            ->getParentMetadataForProperty($containerEntityClass, 'embeddedEntity.plainField');
+        $this->assertEquals($metadata->fieldMappings[$lastPropertyName]['type'], 'boolean');
+    }
+
+    public function getContainerEntityMetadata($class)
+    {
+        $containerEntityClass = 'Sonata\DoctrineORMAdminBundle\Tests\Fixtures\Entity\ContainerEntity';
+        $associatedEntityClass = 'Sonata\DoctrineORMAdminBundle\Tests\Fixtures\Entity\AssociatedEntity';
+        $embeddedEntityClass = 'Sonata\DoctrineORMAdminBundle\Tests\Fixtures\Entity\Embeddable\EmbeddedEntity';
+
+        $metadata = new ClassMetadata($class);
+
+        if ($class === $containerEntityClass) {
```

Comment:
instead of this if conditions i would like to have 3 methods, what do you think?